cocos/scene-graph/prefab/utils.ts

```diff
@@ -84,7 +84,7 @@ export function createNodeWithPrefab (node: Node) {
 
     // restore preserved props
     node._objFlags = _objFlags;
-    node._parent = _parent;
+    node.parent = _parent;
```

Comment:
This change modify the original logic `node.parent = _parent` setter will do more logic than `node._parent = parent`.

```js
set parent (value) {
        this.setParent(value);
    }
```

Reply:
Yep. I should fix it. It is wired that setter of `_parent` has the same logic as `setParent`.

Reply:
How about naming it with `changeParent` or `replaceParent`?
`modifyParent` seems to modify some content of its parent, not just replace it or change it.

Reply:
Modify is more suitable. I got the name from chatgpt.